File: 프로그래머스/unrated/136798. 기사단원의 무기/기사단원의 무기.js
// 기사단원의 무기

function solution(number, limit, power) {
  // 약수의 개수를 구하는 함수
  const getDivisors = (number) => {
    let divisors = 0;
    for (let i = 1; i <= Math.sqrt(number); i++) {
      if (i ** 2 === number) {
        divisors += 1;
      } else if (number % i === 0) {
        divisors += 2;
      }
    }
    return divisors;
  };

  const knight = [];
  for (let i = 1; i <= number; i++) {
    const getDivisor = getDivisors(i);
    if (getDivisor > limit) {
      knight.push(power);
    } else {
      knight.push(getDivisor);
    }
  }

  return knight.reduce((sum, num) => sum + num);
}

File: 프로그래머스/unrated/136798. 기사단원의 무기/기사단원의 무기.py
import math

def check(n):
    tmp = 0
    for i in range(1, int(math.sqrt(n)) + 1):
        if n**0.5 == i:
            tmp += 1
        elif n % i == 0:
            tmp += 2
    return tmp

def solution(number, limit, power):
    result = []
    
    for num in range(1, number + 1):
        target = check(num)
        if target > limit:
            result.append(power)
        else:
            result.append(target)
    
    
    return sum(result)